_adr/87/index.adoc

```diff
@@ -1,48 +1,442 @@
+
+We currently expose two different types of customer facing metrics:
+
+* Raw Kafka metrics -- exposed by Kafka instances.
+* Recording rules -- configured by us for any Kafka metrics requiring pre-computed aggregation.
+
+This results in multiple problems:
+
+* *Consistency*: These two types of metrics follow different naming schemes, resulting in inconsistent metric names.
+* *Repetition*: The raw Kafka metrics contain a lot of repetition i.e. `kafka_controller_kafkacontroller...`.
+* *Exposure of implementation details*: The name of the recording rules metrics exposes implementation details i.e. `kafka_namespace:haproxy_server_bytes_in_total:rate5m`, end users do not need to know about namespaces and OpenShift.
+
+Since we don't have a formal process and naming policy for adding customer facing metrics, we often reach a situation where the resulting metric names are confusing and misleading (i.e. https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/issues/688[misleading metric name for kafka topic partitions count]).
+
 
 ## Goals
-// Bulleted list of outcomes that this ADR, if accepted, should help achieve
+- Define a naming policy for existing and new customer facing metrics.
+- Define a process for requesting new customer facing metrics to the API.
+- Define a process for migrating the existing customer facing metrics to follow the new naming policy.
+
 
 ## Non-goals
-// Bulleted list of outcomes that this ADR is not trying to achieve.
+- Propose solution for metrics API versioning.
+We need to consider KAS Fleet Manager API as a whole in order to have consistent versioning strategy for all of our endpoints, including customer facing metrics and the metrics API.
+This should be separated from this ADR.
+
 
 ## Current situation
-// Where are we now?
+### Requesting new customer facing metrics
+There is currently no formal process that we follow when a new metric is requested to be added to the API. 
+
+The request is handled by the Control Plane team to add the metrics to the KAS Fleet Manager API as well as adding new recording rules to the Observability configuration if needed.
+However, the Kafka Integrations team would have the knowledge of what metrics to use and how it should be aggregated, if required.
+All discussions around which metric to use and how its aggregated is done via ad-hoc chat conversations, in the JIRA issue or the Github pull request.
+Discussion to reach a formal agreement cross teams for the metric name is not done.
+
+
+### Naming scheme
+The names of the current customer facing metrics are based on the following:
+
+* Raw Kafka metrics: The names of these metrics are not changed, they are taken directly from Kafka metrics. These have the following name format:
+** For example: `kafka_controller_kafkacontroller_offline_partitions_count`
```

Comment:
@k-wall defining the correct names for the metrics in the Kafka CR instead of renaming them later via rules in the resources repository seems preferable, would you agree?

Reply:
Yes, it is possible to do that at source

Reply:
Would an epic brief be the right place to discuss/have proposed solutions for how we may do this? @pb82 @k-wall @tombentley

Reply:
Yes, Epic Brief would be the right place.

Reply:
Epic brief has been created for this epic. It's linked in the description of the epic issue in JIRA: [MGDSTRM-7080](https://issues.redhat.com/browse/MGDSTRM-7080)